common/traits/fixed-supply-token/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+#![no_std]
+
+elrond_wasm::imports!();
+
+pub trait FixedSupplyToken<M: ManagedTypeApi> {
+    fn get_total_supply(&self) -> &BigUint<M>;
+
+    fn into_part(self, payment_amount: &BigUint<M>) -> Self;
+
+    /// full_value * current_supply / total_supply
+    fn rule_of_three(&self, current_supply: &BigUint<M>, full_value: &BigUint<M>) -> BigUint<M> {
+        let total_supply = self.get_total_supply();
+        if current_supply == total_supply {
```

Comment:
Why the extra verification? Line 14 would result in the same value.

Reply:
Since this is the most common case (rarely do people actually split positions), it acts as a minor optimization (comparison instead of multiplication and division on potentially very big numbers)